### PR TITLE
API diff between .NET 8 Preview 7 and .NET 8 RC 1

### PR DESCRIPTION
**Pull request by carlossanlop**
**Known tooling issues**:
- The AsmDiff tool sometimes skips adding `{` or `{}`. Please provide a GitHub suggestion in this PR so I can correct it. [arcade#10981](https://github.com/dotnet/arcade/issues/10981)
- The AsmDiff tool shows APIs with decorating attribute changes as removed and re-added, instead of showing only the API. Please ignore this, it might be confusing but technically it's not incorrect. I have an issue to track this: [arcade#13814](https://github.com/dotnet/arcade/issues/13814) 
  
Repos area owners:

- ASP.NET - @dotnet/aspnet-api-review 
- WinForms - @merriemcgaw @dreddy-work @JeremyKuhne
- WPF - @singhashish-wpf and @pchaurasia14
- Libraries - @artl93 @jeffhandley @karelz @ericstj @stephentoub

Libraries area owners:
 
- System @dotnet/area-system-runtime 
- @dotnet/area-system-buffers 
- @dotnet/area-system-collections 
- System.Data @ajcvickers @DavoudEshtehari @David-Engel @roji 
- @dotnet/area-system-io 
- System.Net @dotnet/ncl 
- @dotnet/area-system-numerics 
- @dotnet/area-system-reflection 
- @dotnet/area-system-runtime-compilerservices 
- System.Runtime.InteropServices @dotnet/interop-contrib 
- @dotnet/area-system-runtime-intrinsics
- System.Runtime.Intrinsics.Wasm @lewing @BrzVlad 
- @dotnet/area-system-text-json
- System.Threading @mangod @kouvel

### CODE REVIEW

File: release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Numerics.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# System.Numerics
+
+``` diff
+ namespace System.Numerics {
+     public static class Vector {
+-        public static Vector<T> LoadUnsafe<T>(ref T source);
++        public static Vector<T> LoadUnsafe<T>([RequiresLocationAttribute] ref T source);
```
Comment (stephentoub) [inline release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Numerics.md:7]: Can we search/replace all of these `[RequiresLocationAttribute] ref` with `ref readonly`?

File: release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Buffers.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# System.Buffers
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Buffers.md:1]: System.Buffers LGTM.

File: release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Numerics.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# System.Numerics
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Numerics.md:1]: System.Numerics LGTM.

I agree with Stephen's input that ideally we track `[RequiresLocationAttribute] ref` as `ref readonly`, which is the actual language syntax here.

File: release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Runtime.CompilerServices.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# System.Runtime.CompilerServices
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Runtime.CompilerServices.md:1]: System.Runtime.CompilerServices LGTM.

Similar input around `ref readonly`

File: release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Runtime.Intrinsics.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# System.Runtime.Intrinsics
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Runtime.Intrinsics.md:1]: System.Runtime.Intrinsics LGTM.

Similar input around `ref readonly`

File: release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Runtime.Intrinsics.Wasm.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# System.Runtime.Intrinsics.Wasm
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Runtime.Intrinsics.Wasm.md:1]: CC. @radekdoulik 

Were these APIs intentionally removed?

File: release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Data.Common.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# System.Data.Common
+
+``` diff
+ namespace System.Data.Common {
+     public abstract class DbBatchCommand {
++        public virtual bool CanCreateParameter { get; }
```
Comment (roji) [inline release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Data.Common.md:6]: System.Data LGTM

**Review by roji — APPROVED** [review]

**Review by adamsitnik — APPROVED** [review]
System.IO LGTM, :shipit:

File: release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Runtime.Intrinsics.Wasm.md (hunk 6 above)
Comment (carlossanlop) [reply release-notes/8.0/preview/api-diff/rc1/Microsoft.NETCore.App/8.0-rc1_System.Runtime.Intrinsics.Wasm.md:1]: @radekdoulik @lewing can you take a look?

**Pull request merged**